Userland/Libraries/LibWeb/SVG/SVGStyleElement.cpp

```diff
@@ -0,0 +1,178 @@
+    sheet.set_parent_css_style_sheet(nullptr);
+    sheet.set_owner_node(nullptr);
+    sheet.set_owner_css_rule(nullptr);
+}
+
+// https://www.w3.org/TR/cssom/#add-a-css-style-sheet
+static void add_a_css_style_sheet(DOM::Document& document, CSS::CSSStyleSheet& sheet)
+{
+    // 1. Add the CSS style sheet to the list of document or shadow root CSS style sheets at the appropriate location. The remainder of these steps deal with the disabled flag.
+    document.style_sheets().add_sheet(sheet);
+
+    // 2. If the disabled flag is set, then return.
+    if (sheet.disabled())
+        return;
+
+    // FIXME: 3. If the title is not the empty string, the alternate flag is unset, and preferred CSS style sheet set name is the empty string change the preferred CSS style sheet set name to the title.
+
+    // FIXME: 4. If any of the following is true, then unset the disabled flag and return:
+    //           The title is the empty string.
+    //           The last CSS style sheet set name is null and the title is a case-sensitive match for the preferred CSS style sheet set name.
+    //           The title is a case-sensitive match for the last CSS style sheet set name.
+
+    // FIXME: 5. Set the disabled flag.
+}
+
+// https://www.w3.org/TR/cssom/#create-a-css-style-sheet
+static void create_a_css_style_sheet(DOM::Document& document, DeprecatedString type, DOM::Element* owner_node, DeprecatedString media, DeprecatedString title, bool alternate, bool origin_clean, DeprecatedString location, CSS::CSSStyleSheet* parent_style_sheet, CSS::CSSRule* owner_rule, CSS::CSSStyleSheet& sheet)
+{
+    // 1. Create a new CSS style sheet object and set its properties as specified.
+    // FIXME: We receive `sheet` from the caller already. This is weird.
+
+    sheet.set_parent_css_style_sheet(parent_style_sheet);
+    sheet.set_owner_css_rule(owner_rule);
+    sheet.set_owner_node(owner_node);
+    sheet.set_type(move(type));
+    sheet.set_media(move(media));
+    sheet.set_title(move(title));
+    sheet.set_alternate(alternate);
+    sheet.set_origin_clean(origin_clean);
+    sheet.set_location(move(location));
+
+    // 2. Then run the add a CSS style sheet steps for the newly created CSS style sheet.
+    add_a_css_style_sheet(document, sheet);
+}
+
+// The user agent must run the "update a style block" algorithm whenever one of the following conditions occur:
+// FIXME: The element is popped off the stack of open elements of an HTML parser or XML parser.
+//
+// NOTE: This is basically done by children_changed() today:
+// The element's children changed steps run.
+//
+// NOTE: This is basically done by inserted() and removed_from() today:
+// The element is not on the stack of open elements of an HTML parser or XML parser, and it becomes connected or disconnected.
+//
+// https://html.spec.whatwg.org/multipage/semantics.html#update-a-style-block
+void SVGStyleElement::update_a_style_block()
```

Comment:
I feel like this would be quite easy to share too. It just uses common element functions it seems, so just make it a non-member function (which takes a `DOM::Element&`) and it should be re-usable for HTML/SVG style elements (you could probably add a VERIFY() check the passed element is a <style> element).